ui/lib/components/accordion/accordion.dart

```diff
@@ -126,14 +155,87 @@ class Accordion {
     return _accordionItems.firstWhere((item) => item._id == id, orElse: () => null);
   }
 
+  void _makeSortable() {
+    if (_sortableId == null) return;
+
+    _accordionItems.forEach((accordionItem) => accordionItem.sortableId = _sortableId);
+    var accordionItemRenderElements = _accordionItems.map((accordionItem) => accordionItem.renderElement).toList();
+
+    _draggable?.destroy();
+    _dropzone?.destroy();
+    
+    _draggable = Draggable(accordionItemRenderElements, avatarHandler: AvatarHandler.clone(), verticalOnly: true, handle: '.$ACCORDION_ITEM_HANDLE_CSS_CLASSNAME');
+    _draggable.onDragStart.listen((DraggableEvent event) {
+      collapseAllItems();
+    });
+
+    _dropzone = Dropzone(accordionItemRenderElements);
+    _dropzone.onDragOver.listen((DropzoneEvent event) {
+      event.avatarHandler.avatar?.classes?.toggle('drag-element', true);
+    });
+    _dropzone.onDragEnter.listen((event) {
+      if (_dragAllowed(event)) {
+        event.dropzoneElement.classes.toggle('drag--over', true);
+      }
+    });
+    _dropzone.onDragLeave.listen((event) {
+      if (_dragAllowed(event)) {
+        event.dropzoneElement.classes.toggle('drag--over', false);
+      }
+    });
+    _dropzone.onDrop.listen((DropzoneEvent event) {
+      if (_dragAllowed(event)) {
+        _rearrangeAccordionItems(event.draggableElement, event.dropzoneElement);
+      }
+    });
+  }
+
+  bool _dragAllowed(DropzoneEvent event) {
+    return event.draggableElement.dataset[SORTABLE_DATASET_KEY] == event.dropzoneElement.dataset[SORTABLE_DATASET_KEY];
+  }
+
+  void _rearrangeAccordionItems(Element draggedElement, Element droppedElement) {
```

Comment:
This would be a good example to talk through from our previous discussions in creating a good separation between the controller and the view. What I think we should do here instead is:
- `dropzone.onDrop.listen()` calls something like `onSort(accordionItem, newIndex)` (so very little logic in the UI, we don't generate the expected list, we just pass along the facts of this change)
- `onSort` is just a thin wrapper over `command(UIAction.reorderStandardMessages<Group/Category>, data)`
- command() does the reordering logic and works out which accordion items need to be moved and where, then it updates the index in the model and calls the accordion with the changes that need to appear in the UI

it might feel a bit like the indirection through the controller is unhelpful compared to your current strategy, but 1) by avoiding having to keep track of the list of messages/groups/categories in two places we avoid a pretty difficult to debug set of problems when those two lists get out of sync, and 2) the view simply just displays what the controller knows, meaning that if there's a problem with the data structure in the controller, we get to see it on the screen more clearly than having to debug for it.

Can you give the above approach a try and let me know how you're getting on?